src/update_engine/delta_diff_generator.cc

```diff
@@ -1209,21 +1204,29 @@ bool DeltaDiffGenerator::GenerateDeltaUpdateFile(
     const string& output_path,
     const string& private_key_path,
     uint64_t* metadata_size) {
-  int old_image_block_count = 0, old_image_block_size = 0;
-  int new_image_block_count = 0, new_image_block_size = 0;
-  TEST_AND_RETURN_FALSE(utils::GetFilesystemSize(new_image,
-                                                 &new_image_block_count,
-                                                 &new_image_block_size));
+  off_t new_image_size = 0, old_image_size = 0;
+  TEST_AND_RETURN_FALSE(utils::GetDeviceSize(new_image, &new_image_size));
+  if (new_image_size % kBlockSize != 0) {
+    LOG(ERROR) << "Size of " << new_image << " (" << new_image_size
```

Comment:
Just seems to be consistent with this codebase, but they're also shelling out to run dd for sparse file allocation so... yeah.